app/v2/controllers/contribution/page.js

```diff
@@ -302,5 +304,28 @@
 
       return vm.trustedHtmlText;
     }
+
```

Comment:
Two notes: 
- Before calling `loadCampaign()`, check the `localStorage.get("currentCampaign")`. 
- If `currentCampaign.uuid` or `currentCampaign.campaignId` is different than the campaign id in the UI http path, only then load the campaign again.